packtools/sps/validation/alternatives.py

```diff
@@ -1,30 +1,63 @@
+
+    def validate_children(self):
+        for tag in self.obtained_children:
+            if tag not in (self.parent_children_dict.get(self.obtained_parent) or []):
+                yield self.create_validation_response(
+                    expected=self.parent_children_dict.get(self.obtained_parent),
+                    obtained=self.obtained_children,
+                    advice=f"Provide child tags according to the list: {self.parent_children_dict.get(self.obtained_parent)}"
+                )
+
+    def create_validation_response(self, expected, obtained, advice):
+        return format_response(
+            title="Alternatives validation",
+            parent=self.alternative.get("parent"),
+            parent_id=self.alternative.get("parent_id"),
+            item=self.obtained_parent,
+            sub_item="alternatives",
+            validation_type="value in list",
+            is_valid=False,
+            expected=expected,
+            obtained=obtained,
+            advice=advice,
+            data=self.alternative
+        )
+
+    def validation(self):
+        yield from self.validate_parent()
+        yield from self.validate_children()
 
 
 class AlternativesValidation:
-    def __init__(self, xmltree):
+    def __init__(self, xmltree, parent_children_dict=None):
         self.xmltree = xmltree
-        self.alternatives = list(Alternatives(xmltree).alternatives)
+        self.parent_children_dict = parent_children_dict
+        self.alternatives = ArticleAlternatives(xmltree).alternatives()
 
-    def validation(self):
+    def validation(self, parent_children_dict=None):
+        parent_children_dict = parent_children_dict or self.parent_children_dict
+        if not parent_children_dict:
+            raise ValidationAlternativesException("Function requires dict of parent tag (key) and child tags list (value)")
         for alternative in self.alternatives:
-            alternative_parent = alternative.get("alternative_parent")
-            alternative_children = alternative.get("alternative_children")
-            for tag in alternative_children:
-                if tag not in ALTERNATIVES.get(alternative_parent):
-                    yield format_response(
-                        title="Alternatives validation",
-                        parent=alternative.get("parent"),
-                        parent_id=alternative.get("parent_id"),
-                        item=alternative_parent,
-                        sub_item="alternative",
-                        validation_type="match",
-                        is_valid=False,
-                        expected=ALTERNATIVES.get(alternative_parent),
-                        obtained=alternative_children,
-                        advice=f"Provide child tags according to the list: {ALTERNATIVES.get(alternative_parent)}",
-                        data=alternative
-                    )
+            yield from AlternativeValidation(alternative, parent_children_dict).validation()
```

Comment:
@Rossi-Luciano no lugar de `parente_children_dict`, já tem que estar filtrado para o alternative correspondente. Ou seja, deveria ser algo como:

```python
yield from AlternativeValidation(alternative, parent_children_dict[alternative.getparent().tag]).validation()
```

Mas note que `parent_children_dict[alternative.getparent().tag]` pode levantar exceção se não existir no dicionário, similar a ValidationAlternativesException("Function requires dict of parent tag (key) and child tags list (value)")

Só que lembra que o seu usuário não é desenvolvedor, as mensagens não podem ter os termos para desenvolvedor. Use os termos para o usuário que configurará os valores